spec/mongoid/attributes/nested_spec.rb

```diff
@@ -4249,6 +4249,43 @@ class BandWithAllowDestroyedRecords < Band
           end
         end
       end
+
+      context "when the relationship is has many" do
+
+        context "when a nested attribute is set and allow_destroy is true, and cache counter and touch are true in the sub-document" do
+
+          let(:party) do
+            party = Party.create(name: 'Ruby Conference', guests_attributes: [{ name: 'attendee 1'}])
+            party.reload.guests
+            party.guests.to_a
+            party.update_attributes(name: 'Ruby Conference', guests_attributes: [{ id: party.guests.first, name: 'attendee 1', _destroy: '1'}])
+            party.reload.guests
+            party.guests.to_a
+            party
+          end
+
+          let(:party_timestamps) do
+            party = Party.create(name: 'Ruby Conference', guests_attributes: [{ name: 'attendee 1'}])
+            party.reload.guests
+            party.guests.to_a
+            time_one = party.updated_at
+            sleep 2
+            party.update_attributes(name: 'Ruby Conference', guests_attributes: [{ id: party.guests.first, name: 'attendee 1', _destroy: '1'}])
+            party.reload.guests
+            party.guests.to_a
+            time_two = party.updated_at
+            return time_one, time_two
+          end
+
+          it "updated_at timestamp is updated on destruction of sub-document" do
+            expect(party_timestamps[0]).to be < party_timestamps[1]
```

Comment:
It will be better IMHO to move almost whole let into the actual spec (it block).